src/main/java/com/websummarizer/Web/Summarizer/controller/PasswordResetController.java

```diff
@@ -119,10 +147,9 @@ public String reset(
             HttpServletResponse response
     ) {
         User temp = userService.getUserByEmailAndResetToken(email, code);
-
         if (temp != null) {
             temp.setPassword(password);
-            userService.setPassword(temp);
+            authenticationService.registerUser(temp);
```

Comment:
instead of making an authentication service object, try to use authentication controller, since no user should be able to communicate with services.

Reply:
I've committed the change to github. What about my usage of the user service object? Is that fine to keep as is?